src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl() {
+        operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(Operation.BALANCE, new BalanceOperation());
+        operationHandlerMap.put(Operation.PURCHASE, new PurchaseOperation());
+        operationHandlerMap.put(Operation.RETURN, new ReturnOperation());
+        operationHandlerMap.put(Operation.SUPPLY, new SupplyOperation());
+    }
```

Comment:
Instantiating the `OperationHandler` implementations directly within the `OperationStrategyImpl` constructor can lead to issues if these handlers have state or require dependencies. It's better to inject these dependencies through the constructor, allowing for easier testing and management of the handler instances. Consider using a dependency injection framework or passing a map of handlers when constructing an `OperationStrategyImpl` instance.